### PR TITLE
allow .env style output from resolve and add associated docs

### LINKED ISSUE #160: [Feature Request] better service selection in subshell
### Description

When running the dmno cli within a subshell - for example `supabase secrets set --env-file < (pnpm exec dmno resolve --format env)`, we cannot rely on any interactive user prompts.

When we can't infer the current service from the package manager, we rely on `cwd`, but we just set the selection in an interactive prompt and let the user confirm it - which is problematic in a subshell.

### Motivation

having the dmno cli hang within a subshell is not great

### Proposed Solution

Currently  the `dev` and `init` commands support a `--silent` flag which will actually select the service and skip the prompt, but `resolve` has no such flag.

We can:
- add the `--silent` flag to `dmno resolve`
- detect that we are in a subshell and enable similar behaviour automatically

generally, we just need to explore what info we get when running dmno within a subshell and try to do the right thing.

### Alternatives

users can explicitly pass a service selection `-s myservice` but it's not obvious since it silently hangs

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by philmillman**
Adds the `env` format option for the `--format` flag in `dmno resolve`
Adds the -np or --no-prompt flag to bypass the service selection for `dmno resolve`
Automatically detects if you're in a non-interactive shell (like a subshell) and also moves into non-interactive mode

For example: 

`pnpm exec dmno resolve --service api --format env >> .env.production`

adds docs to: 
- `dmno resolve` page
- .env Files guide page

fixes #160

### CODE REVIEW

**Review by theoephraim — APPROVED** [review]
awesome - just needs a changeset!

File: packages/core/src/cli/lib/shell-helpers.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const isSubshell = () => (process.env.BASH_SUBSHELL && process.env.BASH_SUBSHELL !== '0') || false;
```
Comment (theoephraim) [inline packages/core/src/cli/lib/shell-helpers.ts:1]: simple enough! Any other notes about other things you tried / learned worth adding as comments?

File: packages/core/src/cli/lib/shell-helpers.test.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { execSync, spawn } from 'child_process';
+import { expect, test, describe } from 'vitest';
+import { isSubshell } from './shell-helpers';
+
+describe('isSubshell', () => {
```
Comment (theoephraim) [inline packages/core/src/cli/lib/shell-helpers.test.ts:5]: hooray tests! 🥳

**Review by theoephraim — APPROVED** [review]

File: packages/core/src/cli/lib/shell-helpers.ts (hunk 1 above)
Comment (philmillman) [reply packages/core/src/cli/lib/shell-helpers.ts:1]: I think the _better_ way is probably to verify via that the pid differs from the parent pid, but this is different across shells (zsh for example)

**Pull request merged**